projects/project-ideas.md

```diff
@@ -106,10 +106,18 @@ Paprika - new C# DB optimized for Merkle Tries and Keccaks
 
 ### Auditing Beacon APIs in Prysm
 
-By Radek
+Currently it's not possible to connect a Prysm validator client (VC) to a non-Prysm beacon node. Before the third cohort started, our VC could interact with the beacon node (BN) only through a Prysm-specific set of gRPC endpoints. EPF's third cohort participants set out on a journey to change this and allow the VC to communicate over HTTP using  Beacon APIs. They got very close to making it happen, and it's probably a matter of weeks before we have a working implementation. That being said, there's still a lot to be done before we can safely advice our users to turn the feature on.
+
+When Beacon APIs were initially implemented, we mostly focused on correctness and didn't pay attention to efficiency. Because of this some endpoints are very unoptimized. They contain expensive loops and don't make use of certain caches. On top of this, server-side handlers are written as gRPC functions, which results in having a middleware layer to convert between HTTP and gRPC. This is very costly. On the contrary, Prysm-specific gRPC endpoints are optimized to make the VC <--> BN communication as fast as possible. We want to eventually deprecate these endpoints in favor of Beacon APIs, but we can't do that if we have to sacrifice too much.
```

Comment:
```suggestion
When initially implementing Beacon APIs, our primary focus was on correctness rather than efficiency. As a result some endpoints are very unoptimized. They contain expensive loops and don't make use of certain caches. Additionally, server-side handlers are written as gRPC functions, necessitating a middleware layer to convert between HTTP and gRPC. This is very costly. On the contrary, Prysm-specific gRPC endpoints are optimized to make the VC <--> BN communication as fast as possible. We want to eventually deprecate these endpoints in favor of Beacon APIs, but we can't do that if we have to sacrifice too much.
```